### ISSUE TITLE
F.c.book/iOS rendering broken

### ISSUE THREAD

**Issue by ascheman** [dukecon/dukecon#13]
It seems the built-in Facebook browser on iOS cannot show DukeCon, cf. screen shot below.

_Hint_: It seems to be only a problem with some iOS versions (in particular with _iPhone7_ ;-)? See below! We need to collect more data about the issue.
## How to reproduce the problem

If you would like to help investigate the problem, pls. do the following
- Open the Facebook App on your iPhone or iPad
- Search for Javaland
- Open the conference web site within the Facebook browser
- Search for a link to the conference program
- Click it!
- If it does not show the talk list but an empty screen like the in the first screen shot below, do the following:
  - Note current time and day
  - Check your current IP address by opening: 
  - Search for the current version of your iOS (somewhere in the settings, you will find some screen shots below)
  - Make a screen shot of your iOS version info
  - Send all of this information (date/time, IP, version info) to feedback@dukecon.org or do directly add a comment here.
    Thx a lot!
## Solution ideas

A first work around should be to check for the browser header and show a special page for the user that there will be rendering problems within the Facebook app on iOS and to recommend to switch to a "real" web browser.
## Further information

The problem is [well known for Facebook@iOS](http://fieldguide.gizmodo.com/how-to-open-links-in-your-mobile-browser-instead-of-the-1657320067). 
### Problem screen shot

![img_5090](https://cloud.githubusercontent.com/assets/456000/10071756/8bed9c34-62bc-11e5-9575-9f653a8d227a.png)

**Comment by sippsack** [comment]
Browser-Kennung:
http://www.ophidia.net/insane/ios-facebook-app-browser-erkennen/

Abschalten In-App-Browser:
http://www.netzwelt.de/news/151781-facebook-app-app-browser-deaktivieren.html

**Comment by ascheman** [comment]
Oops, sometimes it seems to work!!!???
![image1](https://cloud.githubusercontent.com/assets/456000/10082061/32be110c-62f8-11e5-9a3a-7a439765b267.PNG)
![image1](https://cloud.githubusercontent.com/assets/456000/10082067/3907776a-62f8-11e5-9497-28b24afffc18.PNG)

Maybe we should only check for certain Fakebook browser versions? This one had `[FBAN/FBIOS;FBAV/39.1.0.41.31;FBBV/14951342;FBDV/iPhone5,2;FBMD/iPhone;FBSN/iPhone OS;FBSV/9.0;FBSS/2; FBCR/Vodafone.de;FBID/phone;FBLC/de_DE;FBOP/5]`

**Comment by annam002** [comment]
Same here; IPhone Version 8.4.

**Comment by ascheman** [comment]
@annam002 you seem to have `FBAN/FBIOS;FBAV/39.1.0.41.31;FBBV/14951342;FBDV/iPhone5,2;FBMD/iPhone;FBSN/iPhone OS;FBSV/8.4;FBSS/2; FBCR/Telekom.de;FBID/phone;FBLC/de_DE;FBOP/5`

**Comment by ascheman** [comment]
Facebook update on the very same iPhone as of 19:44 does not change the browser id:  `FBAN/FBIOS;FBAV/39.1.0.41.31;FBBV/14951342;FBDV/iPhone5,2;FBMD/iPhone;FBSN/iPhone OS;FBSV/9.0;FBSS/2; FBCR/Vodafone.de;FBID/phone;FBLC/de_DE;FBOP/5`

**Comment by ascheman** [comment]
New problem report from IP 91.65.95.35 with Browser ID `FBAN/FBIOS;FBAV/39.1.0.41.31;FBBV/14951342;FBDV/iPhone7,2;FBMD/iPhone;FBSN/iPhone OS;FBSV/9.0;FBSS/2; FBCR/Telekom.de;FBID/phone;FBLC/de_DE;FBOP/5`

The only difference I see so far is _iPhone7_ vs. _iPhone5_

![img_2282](https://cloud.githubusercontent.com/assets/456000/10094808/b01ce9ea-6364-11e5-92f8-e21ad6540e69.PNG)

**Comment by ascheman** [comment]
OK, my colleague has the same problem:
- IP 80.187.98.196 (9:32)
- iOS 9.0.1!
- `FBAN/FBIOS;FBAV/39.1.0.41.31;FBBV/14951342;FBDV/iPhone7,2;FBMD/iPhone;FBSN/iPhone OS;FBSV/9.0.1;FBSS/2; FBCR/Telekom.de;FBID/phone;FBLC/en_US;FBOP/5`
- Obviously, the _iPhone6_ is an _iPhone7_ ;-)

![img_3719](https://cloud.githubusercontent.com/assets/456000/10096387/5ea05ec8-6371-11e5-974d-fcc3c515a753.PNG)

**Comment by annam002** [comment]
Could reproduce the problem on a different browser and committed a bug fix. Maybe this will also fix the issue for the other browsers. Tests will show

**Comment by ascheman** [comment]
Latest change of @annam002 (dukecon/dukecon_html5@76a813fe81830fdb2dc156d63b8798e46ec44871) improved situation on iPhone6 with Facebook-Browser: Talks come up now. 

However, offline mode on iPhone 6 with FB-App/-Browser is still not working (time out after 30 seconds, see screen shot below).

![img_3821](https://cloud.githubusercontent.com/assets/456000/10132275/f7c86b5c-65d5-11e5-96a8-2f171ad67f95.PNG)

**Comment by annam002** [comment]
Yeah, it is not surprising that it does not work offline now, because the bug was that it could not access the indexeddb and I just added some checks so that the unavailability of indexeddb does not crash the whole page. 
We should do some research if it is indeed not supported in this browser, or if our code just does not grab the right object from the window. If it indeed does not support indexeddb, I'm not sure that there is a workaround to make it work offline.

**Comment by mescribano** [comment]
Hello everyone,

I reported this bug to Apple Developer Bug Reporting about a month ago.

I got an update yesterday that I should verify this issue with iOS 9.1 GM (Build: 13B143).
As I got this update I had still on my device the version 9.0.2. I tested again before doing the upgrade and with iOS 9.0.2 the bug was still happening.

I updated to iOS 9.1 and after an initial test on my iPhone the bug seems to be fixed. :)

Cheers
